lib/Cro/Iri.pm6

```diff
@@ -203,12 +214,30 @@ class Cro::Iri does Cro::ResourceIdentifier {
             make '';
         }
 
+        method isegment($/) {
+            make $<ipchars>.ast;
+        }
+
+        method isegment-nz($/) {
+            make $<ipchars>.ast;
+        }
+
         method iquery($/) {
-            make ~$/;
+            my $result = '';
+            $result ~= $_<ipchars> ?? $_<ipchars>.ast !! ~$_ for @$0;
+            make $result;
         }
 
         method ifragment($/) {
-            make ~$/;
+            my $result = '';
+            $result ~= $_<ipchars> ?? $_<ipchars>.ast !! ~$_ for @$0;
+            make $result;
+        }
+
+        method ipchars($/) {
+            my $result = '';
+            $result ~= $_<broken> ?? encode-percents(~$_) !! ~$_ for @$0;
```

Comment:
`$_<broken>` is more idimoatically `.<broken>`

Reply:
Yes, but I'd prefer the ease of reading here because after the condition we use `$_` twice, in both branches, so all the three cases are using the same form which is easy to understand.

Reply:
Fair.